File: JavaScript/leetcode_78.js
function subsets(nums) {
  const result = [];

<<<<<<< HEAD
  function backtrack(start, path) {
    result.push([...path]); // Add a copy of current subset

    for (let i = start; i < nums.length; i++) {
      path.push(nums[i]); // Choose
      backtrack(i + 1, path); // Explore
      path.pop(); // Un-choose (backtrack)
=======
    function backtrack(start, path) {
        result.push([...path]); // Add a copy of current subset

        for (let i = start; i < nums.length; i++) {
            path.push(nums[i]); // Choose
            backtrack(i + 1, path); // Explore
            path.pop(); // Un-choose (backtrack)
        }
>>>>>>> a240f511499660ef602885c94b58faa12d2f12e8
    }
  }

<<<<<<< HEAD
  backtrack(0, []);
  return result;
=======
    backtrack(0, []);
    return result;
>>>>>>> a240f511499660ef602885c94b58faa12d2f12e8
}

File: Python/leetcode_78.py

from typing import List


def subsets(nums: List[int]) -> List[List[int]]:
    result = []

    def backtrack(start, path):
        result.append(path[:])  # Add a copy of current subset
        for i in range(start, len(nums)):
            path.append(nums[i])
            backtrack(i + 1, path)
            path.pop()  # Backtrack

    backtrack(0, [])
    return result


print(subsets([1, 2, 3]))
